cpp/ql/src/semmle/code/cpp/Declaration.qll

```diff
@@ -211,6 +211,26 @@ class Declaration extends Locatable, @declaration {
   /** Holds if this declaration is a member of a class/struct/union. */
   predicate isMember() { hasDeclaringType() }
 
+  /** Holds if this declaration is a member and has an lvalue ref-qualifier */
+  predicate isLValueQualified() {
+    isMember() and hasSpecifier("lvalue_ref_qualifier")
+  }
+
+  /** Holds if this declaration is a member and has an rvalue ref-qualifier */
+  predicate isRValueQualified() {
+    isMember() and hasSpecifier("rvalue_ref_qualifier")
+  }
+
+  /** Holds if this declaration is a member and has a ref-qualifier */
+  predicate isRefQualified() {
+    isLValueQualified() or isRValueQualified()
+  }
+
+  /** Holds if this declaration is a member and does not have a ref-qualifier */
+  predicate isUnqualified() {
+    isMember() and not isRefQualified()
+  }
```

Comment:
`const` and `volatile` are also considered qualifiers, so this predicate name is misleading. We could give it a longer name like `isNonRefQualified`. Also, if these get moved to `MemberFunction`, we don't need it at all

Reply:
Agreed, thanks! :)

Reply:
I agree with moving the new predicates to `MemberFunction`, mainly because they'll be used so rarely.